packages/proxy/src/providers/openai.test.ts

```diff
@@ -0,0 +1,182 @@
+import { describe, it, expect } from "vitest";
+import { callProxyV1 } from "../../utils/tests";
+import {
+  OpenAIChatCompletion,
+  OpenAIChatCompletionChunk,
+  OpenAIChatCompletionCreateParams,
+} from "@types";
+
+it("should deny reasoning_effort for unsupported models streaming", async () => {
+  const { json } = await callProxyV1<
+    OpenAIChatCompletionCreateParams,
+    OpenAIChatCompletionChunk
+  >({
+    body: {
+      model: "gpt-4o-mini",
+      reasoning_effort: "high",
+      messages: [
+        { role: "system", content: "You are a helpful assistant." },
+        { role: "user", content: "Tell me a short joke about programming." },
+      ],
+      stream: true,
+      max_tokens: 150,
+    },
+  });
+
+  expect(json()).toEqual({
```

Comment:
this is another good usecase for snapshots but not blocking

Reply:
how do snapshots handle very variable data? i.e. on each request the content of the chat completion may differ or the system_fingerprint may be different